go/signedexchange/cmd/dump-signedexchange/main.go

```diff
@@ -13,42 +14,64 @@ import (
 
 var (
 	flagInput     = flag.String("i", "", "Signed-exchange input file")
-	flagSignature = flag.Bool("signature", false, "Print only signature value")
+	flagURI       = flag.String("uri", "", "Signed-exchange uri")
+	flagSignature = flag.Bool("signature", false, "Print signature value")
 	flagVerify    = flag.Bool("verify", false, "Perform signature verification")
 	flagCert      = flag.String("cert", "", "Certificate CBOR file. If specified, used instead of fetching from signature's cert-url")
+	flagHeaders   = flag.Bool("headers", true, "Print headers")
+	flagPayload   = flag.Bool("payload", true, "Print payload")
 )
 
 func run() error {
-	in := os.Stdin
+	var e *signedexchange.Exchange
 	if *flagInput != "" {
-		var err error
-		in, err = os.Open(*flagInput)
+		in, err := os.Open(*flagInput)
 		if err != nil {
 			return err
 		}
 		defer in.Close()
+		e, err = signedexchange.ReadExchange(in)
+		if err != nil {
+			return err
+		}
+	} else if *flagURI != "" {
+		client := http.DefaultClient
+		req, err := http.NewRequest("GET", *flagURI, nil)
+		req.Header.Add("Accept", "application/signed-exchange;v=b3")
+		req.Header.Add("AMP-Cache-Transform", "google;v=1")
+		resp, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		e, err = signedexchange.ReadExchange(resp.Body)
+		if err != nil {
+			return err
+		}
 	}
+	if e != nil {
+		if *flagHeaders {
+			e.PrettyPrintHeaders(os.Stdout)
+		}
 
-	e, err := signedexchange.ReadExchange(in)
-	if err != nil {
-		return err
-	}
+		if *flagPayload {
+			e.PrettyPrintPayload(os.Stdout)
+		}
 
-	if *flagSignature {
-		fmt.Println(e.SignatureHeaderValue)
-		return nil
-	}
+		if *flagSignature {
+			fmt.Println(e.SignatureHeaderValue)
+		}
 
-	if *flagVerify {
-		if err := verify(e); err != nil {
-			return err
+		if *flagVerify {
+			fmt.Println()
+			if err := verify(e); err != nil {
+				return err
+			}
 		}
-		fmt.Println()
+	} else {
+		fmt.Println("Need to pass -uri with URI of sxg or -i with file location of sxg")
```

Comment:
I use `dump-signedexchange` at the end of a pipe sometimes. I'd prefer if you updated this to allow the case where stdin != /dev/pty or whatever. Alternatively, maybe we could make `-i -` a valid syntax if it isn't already.

Reply:
I didn't know you could use `dump-signedexchange` like that! Makes a lot of sense though. I've added that functionality back and documented a use case in the README.

Reply:
Nice! Thanks.